src/collection/collectionview.cpp

```diff
@@ -545,6 +549,41 @@ void CollectionView::OpenInNewPlaylist() {
 
 }
 
+void CollectionView::SearchForThis() {
+
+  QModelIndex current = currentIndex();
+  QVariant type = model()->data(current, CollectionModel::Role_Type);
+  if (!type.isValid() || (type.toInt() != CollectionItem::Type_Song && type.toInt() != CollectionItem::Type_Container && type.toInt() != CollectionItem::Type_Divider)) {
+    return;
+  }
+  QString search;
+
+  switch (type.toInt()) {
+    case CollectionItem::Type_Song:{
+      QModelIndex index = qobject_cast<QSortFilterProxyModel*>(model())->mapToSource(current);
+      SongList songs = app_->collection_model()->GetChildSongs(index);
+      if (!songs.isEmpty()) {
+        last_selected_song_ = songs.last();
+      }
+      search = "title:" + last_selected_song_.title();
+      break;
+    }
+
+    case CollectionItem::Type_Container:
+    case CollectionItem::Type_Divider:{
+      // We could try to assess the GroupBy settings to add a field name, but
+      // not all GroupBy combinations are supported search keywords
+      search = model()->data(current, CollectionModel::Role_Key).toString();
```

Comment:
The key is not what we want here, that's just used internally to identify the unqiue container item. `Qt::DisplayRole` is what we want here.
And it would nice if we could resolve the group by based on `container_level` and prefix "albumartist:", "artist:" and "album:", if it's not a known search prefix, just use the display text. I think you can get the item with `app_->collection_model()->IndexToItem(index)`.

Reply:
I've gone ahead and added field prefixes as requested.

One open question I have is what the users' expectations will be when they "search for this" on a track or an album.

* For a track, will they expect:
  * All tracks with the same title?
  * All tracks by the same artist with the same title (e.g. if the artist has multiple versions of the same song across several albums)?

* For album, will they expect:
  * all albums with the same name?
  * albums by the same artist with the same name (e.g. when the library contains >1 release of the same album)?
  * the exact album (I don't think it's this one, since the search grouping by album won't necessarily yield a unique result by album)?

I think the current approach (no assumptions about further grouping) is sufficient. Between filtering too narrowly and filtering too broadly, I think the latter will be better since the tree view will still allow the user to easily focus on what they're looking for. Thoughts?

Reply:
I'm in favor of doing it the way you've done it, all tracks with the same title, all albums with the same title, etc.